File: backend/app/schemas/breathing.py
"""Pydantic-схеми для вправ на дихання."""

from pydantic import BaseModel, Field


class BreathingPhase(BaseModel):
    """Одна фаза дихальної вправи."""

    phase: str = Field(
        description="Тип фази: inhale | hold | exhale",
        examples=["inhale", "hold", "exhale"],
    )
    label_uk: str = Field(
        description="Опис фази українською",
        examples=["Вдих", "Затримка", "Видих"],
    )
    duration_seconds: int = Field(
        ge=1,
        description="Тривалість фази у секундах",
    )


class BreathingExerciseRead(BaseModel):
    """Повна інформація про дихальну вправу."""

    id: str = Field(description="Унікальний ідентифікатор вправи")
    name_uk: str = Field(description="Назва вправи українською")
    description_uk: str = Field(description="Опис вправи українською")
    phases: list[BreathingPhase] = Field(description="Фази дихального циклу")
    cycles: int = Field(ge=1, description="Кількість циклів")
    total_duration_seconds: int = Field(
        ge=1, description="Загальна тривалість вправи у секундах"
    )
    difficulty: str = Field(
        description="Складність: easy | medium | hard",
        examples=["easy", "medium", "hard"],
    )
    tags: list[str] = Field(description="Теги вправи")


class BreathingExerciseShort(BaseModel):
    """Коротка інформація про дихальну вправу (для списку)."""

    id: str
    name_uk: str
    description_uk: str
    cycles: int
    total_duration_seconds: int
    difficulty: str
    tags: list[str]

File: frontend/src/api/breathing.ts
import client from './client';

/* ── Types ── */

export interface BreathingPhase {
  phase: 'inhale' | 'hold' | 'exhale';
  label_uk: string;
  duration_seconds: number;
}

export interface BreathingExercise {
  id: string;
  name_uk: string;
  description_uk: string;
  phases: BreathingPhase[];
  cycles: number;
  total_duration_seconds: number;
  difficulty: 'easy' | 'medium' | 'hard';
  tags: string[];
}

export interface BreathingExerciseShort {
  id: string;
  name_uk: string;
  description_uk: string;
  cycles: number;
  total_duration_seconds: number;
  difficulty: 'easy' | 'medium' | 'hard';
  tags: string[];
}

/* ── API ── */

export async function getBreathingExercises(): Promise<BreathingExerciseShort[]> {
  const res = await client.get<BreathingExerciseShort[]>('/breathing-exercises/');
  return res.data;
}

export async function getBreathingExercise(id: string): Promise<BreathingExercise> {
  const res = await client.get<BreathingExercise>(`/breathing-exercises/${id}`);
  return res.data;
}

